### PR TITLE
Add documentation for webauthn auth provider configuration

### PR DESCRIPTION
**Pull request by VDigitall**
## Proposed change

Add documentation for webauthn auth provider configuration

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
   - https://github.com/home-assistant/core/pull/122725
   - https://github.com/home-assistant/frontend/pull/21485
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced a new authentication provider, "Webauthn," enhancing security and convenience through passkey login.

- **Documentation**
	- Updated documentation with clear configuration examples for enabling the Webauthn provider.
	- Included detailed descriptions of key configuration parameters to assist users.
	- Emphasized management of passkeys through the Home Assistant user interface and the necessity for compatible hardware and browser support.

### CODE REVIEW

**Comment by c0ffeeca7** [comment]
Thank you for contributing,  @VDigitall :+1:

1. As this PR seems to be related to a code PR, please rebase and point this PR against the `next` branch.
2. Please fill in the PR template.

**Comment by c0ffeeca7** [comment]
I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

File: source/_docs/authentication/providers.markdown [hunk 1]
```diff
@@ -194,3 +194,40 @@ Any leading and trailing whitespace is stripped from usernames before they're pa
 {% note %}
 For now, meta variables are only respected the first time a particular user is authenticated. Upon subsequent authentications of the same user, the previously created user object with the old values is reused.
 {% endnote %}
+
+### Webauthn
+
+The webauthn auth provider allows log in with a passkey. Passkeys are a secure and convenient way to authenticate users using web authentication standards.
+
+To configure the webauthn auth provider, add the following code to your `configuration.yaml` file:
```
Comment (c0ffeeca7) [inline source/_docs/authentication/providers.markdown]:
```suggestion
To configure the webauthn auth provider, add the following code to your {% term "`configuration.yaml`" %} file:
```

File: source/_docs/authentication/providers.markdown [hunk 2]
```diff
@@ -194,3 +194,40 @@ Any leading and trailing whitespace is stripped from usernames before they're pa
...
+  auth_providers:
+    - type: homeassistant
+    - type: webauthn
+      rp_id: localhost
+      expected_origin: http://localhost:8123
+```
+
+{% configuration %}
+rp_id:
+  description: The `rp_id` (relying party identifier) is a unique identifier for the relying party (i.e., the server) that is relying on the WebAuthn authentication. It is used to ensure that the authentication request is intended for the correct server. The `rp_id` can be a domain name or an IP address. For example, if the `rp_id` is set to "example.com", the WebAuthn authentication will only be valid for requests originating from that domain.
```
Comment (c0ffeeca7) [inline source/_docs/authentication/providers.markdown]:
```suggestion
  description: The `rp_id` (relying party identifier) is a unique identifier for the relying party (for example, the server) that is relying on the WebAuthn authentication. It is used to ensure that the authentication request is intended for the correct server. The `rp_id` can be a domain name or an IP address. For example, if the `rp_id` is set to "example.com", the WebAuthn authentication will only be valid for requests originating from that domain.
```

File: source/_docs/authentication/providers.markdown [hunk 3]
```diff
@@ -194,3 +194,40 @@ Any leading and trailing whitespace is stripped from usernames before they're pa
...
+  required: true
+  type: string
+expected_origin:
+  description: The URL from which the user is accessing the login page for webauthn authentication.
+  required: true
+  type: string
+rp_name:
+  description: The name of the relying party (i.e., the server) that is relying on the WebAuthn authentication.
```
Comment (c0ffeeca7) [inline source/_docs/authentication/providers.markdown]:
```suggestion
  description: The name of the relying party (for example, the server) that is relying on the WebAuthn authentication.
```

File: source/_docs/authentication/providers.markdown [hunk 4]
```diff
@@ -194,3 +194,40 @@ Any leading and trailing whitespace is stripped from usernames before they're pa
...
+  required: false
+  default: "Home Assistant"
+  type: string
+{% endconfiguration %}
+
+Once configured, users will be able to log in using their webauthn passkeys. Passkeys can be registered and managed through the Home Assistant user interface.
+
+Please note that the webauthn auth provider requires compatible hardware and browser support. Make sure your devices and browsers support webauthn before enabling this authentication method.
+
+Managing passkeys is available through the UI `User Profile -> Security`
```
Comment (c0ffeeca7) [inline source/_docs/authentication/providers.markdown]:
```suggestion
You can manage your passkeys in the UI. Go to {% my profile title="**User profile**" %} and open the  **Security** tab.
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]
Tiny style tweaks

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @VDigitall :+1:

**Comment by gonzotek** [comment]
The core PR mentions hardware security keys(e.g. YubiKeys).  Should the documentation also mention them in addition to passkeys?  PS I'm hopeful this feature is added soon, I ❤️ passkeys :-)

**Comment by VDigitall** [comment]
> The core PR mentions hardware security keys(e.g. YubiKeys).  Should the documentation also mention them in addition to passkeys?  PS I'm hopeful this feature is added soon, I ❤️ passkeys :-)

I guess yes, I’ll update the documentation

**Pull request closed without merging**